cmd/rekor-cli/app/pflags.go

```diff
@@ -56,8 +57,10 @@ var pflagValueFuncMap map[FlagType]newPFlagValueFunc
 func initializePFlagMap() {
 	pflagValueFuncMap = map[FlagType]newPFlagValueFunc{
 		uuidFlag: func() pflag.Value {
-			// this corresponds to the merkle leaf hash of entries, which is represented by a 64 character hexadecimal string
-			return valueFactory(uuidFlag, validateString("required,len=64,hexadecimal"), "")
+			// this corresponds to the merkle leaf hash of entries, which is represented by a 64 character hexadecimal string, prepended by the 6-digit shard ID and '-'  separator
+			// TODO: does this have to be a literal? can we not use the const to define the len?
+			// TODO: is there a way to validate only the UUID part of this string as hexadecimal? or validate different parts of the string differently? can / should they be split into different flags?
```

Comment:
you could make a custom validator! instead of `validateString` you could make `validateShardedUUID` with a custom impl like the ones found here https://github.com/sigstore/rekor/blob/5025a24e2861d1876d67a09cf6bfad0ee76d64c8/cmd/rekor-cli/app/pflags.go#L170

Reply:
Thank you @asraa ! That's exactly what I want to do. I appreciate the example.